File: tests/xcalablemp/NOT-IMPLEMENTED/template_fix/C/template_fix1.c
#include<xmp.h>
#include<mpi.h>
#include<stdio.h>
#include<stdlib.h>     
#pragma xmp nodes p(*)
#pragma xmp template t(:)
#pragma xmp distribute t(block) onto p
int i,N,s;
int *a;
#pragma xmp align a[i] with t(i)
char *result;
int main(void){

   N=0;
   for(i=1;N<1001;i++)
      N+=1;
 
#pragma xmp temlate_fix(block) t(N)
   a=(int *)malloc(sizeof(int) * N);

#pragma xmp loop (i) on t(i)
   for(i=0;i<N;i++){
      a[i]=i;
   }

   s=0;
#pragma xmp loop (i) on t(i) reduction(+:s)
   for(i=0;i<N;i++){
      s+=a[i];
   }

   result="OK";
   if(s != 499500){
      result = "NG";
   }

   printf("%d %s %s\n",xmp_all_node_num(),"testp001.c",result);
   free(a); 
   return 0;
}
      
         
      
   

File: tests/xcalablemp/NOT-IMPLEMENTED/template_fix/F/template_fix1.f90
      program main
      include 'xmp_lib.h'
!$xmp nodes p(*)
!$xmp template t(:)
!$xmp distribute t(block) onto p
      integer N, s
      integer,allocatable:: a(:)
!$xmp align a(i) with t(i)
      character(len=2) result

      N = 0
      do i=1, 1000
         N = N+1
      enddo
      
!$xmp template_fix (block) t(N)
      allocate(a(N))

!$xmp loop (i) on t(i)
      do i=1, N
         a(i) = i
      enddo
         
      s = 0
!$xmp loop (i) on t(i) reduction(+: s)
      do i=1, N
         s = s+a(i)
      enddo

      result = 'OK'
      if(s .ne. 500500) then
         result = 'NG'
      endif

      print *, xmp_all_node_num(), 'testp001.f ', result
      deallocate(a)

      end
